### PR TITLE
[5.3] On ELF platforms, remove the host toolchain rpath from executables and shared libraries

### PR DESCRIPTION
**Pull request by buttaface**
before installing.

It has been suggested to me by some Swift devs that this would be good to get in before the 5.3 release, a cherry-pick from #2703.

This should be completely safe for the 5.3 branch, as it's just removing the build toolchain's stdlib rpath, which should be unused by SPM.

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by tomerd** [comment]
cc @drexin @compnerd

**Comment by compnerd** [comment]
I think that this is pretty important - it should remove hardcoded paths from the build host which are a problem, we should try to get this into the 5.3 release even though we are cutting it really close.

**Comment by tomerd** [comment]
this looks like a good improvement, we will put into an update as soon as possible

**Comment by buttaface** [comment]
Sounds good.

**Comment by neonichu** [comment]
Looks like this will need an update because of conflicts in the bootstrap script that we introduced. Sorry about that!

**Comment by buttaface** [comment]
No problem, an easy fix.

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by rballard** [comment]
We've shipped 5.3 and are about to start taking fixes for the next patch release. Let's hold this one until Monday, but I expect to approve then.

**Review by rballard — APPROVED** [review]
This is approved to merge now; thank you!

**Pull request merged**
